packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -164,13 +180,17 @@ export const ThemeProvider = React.forwardRef((props, forwardedRef) => {
   themeOptions.highContrast ??=
     themeProp === 'inherit' ? parent.highContrast : undefined;
 
+  const future = futureProp ?? {};
+  future.themeBridge ??=
+    themeProp === 'inherit' ? parent.context?.future?.themeBridge : undefined;
+
   const [portalContainerFromParent] = useScopedAtom(portalContainerAtom);
 
   const contextValue = React.useMemo(
-    () => ({ theme, themeOptions }),
+    () => ({ theme, themeOptions, future }),
     // we do include all dependencies below, but we want to stringify the objects as they could be different on each render
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [theme, JSON.stringify(themeOptions)],
+    [theme, JSON.stringify(themeOptions), future],
```

Comment:
By "different" I mean literally different (i.e. fails the `===` check). `{ themeBridge: true }` is an object literal which gets re-created on every render (unless it's already memoized when passed). You can test this out in the browser console.

Using `JSON.stringify` ensures that the contents of the object are compared, so that our memoized context value only invalidates when the options are actually different.